qiskit/transpiler/preset_passmanagers/plugin.py

```diff
@@ -0,0 +1,256 @@
+     - ``qiskit.transpiler.scheduling``
+     - ``alap``, ``asap``
+     - This is the last stage run and it is expected to output a scheduled
+       circuit such that all idle periods in the circuit are marked by explicit
+       :class:`~qiskit.circuit.Delay` instructions.
+
+Writing Plugins
+===============
+
+To write a pass manager stage plugin there are 2 main steps. The first step is
+to create a subclass of the abstract plugin class
+:class:`~.PassManagerStagePluginManager` which is used to define how the :class:`~.PassManager`
+for the stage will be constructed. For example, to create a ``layout`` stage plugin that just
+runs :class:`~.VF2Layout`::
+
+    from qiskit.transpiler.preset_passmanagers.plugin import PassManagerStagePlugin
+    from qiskit.transpiler.preset_passmanagers import common
+    from qiskit.transpiler import PassManager
+    from qiskit.transpiler.passes import VF2Layout
+
+    class VF2LayoutPlugin(PassManagerStagePlugin):
+
+        def pass_manager(self, pass_manager_config):
+            layout_pm = PassManager([VF2Layout(
+                coupling_map=pass_manager_config.coupling_map,
+                properties=pass_manager_config.backend_properties,
+                target=pass_manager_config.target
+            )])
+            layout_pm += common.generate_embed_passmanager(pass_manager_config.coupling_map)
+            return layout_pm
+
+The second step is to expose the :class:`~.PassManagerStagePluginManager`
+subclass as a setuptools entry point in the package metadata. This can be done
+by simply adding an ``entry_points`` entry to the ``setuptools.setup`` call in
+the ``setup.py`` or the plugin package with the necessary entry points under the
+appropriate namespace for the stage your plugin is for. You can see the list
+of stages, entrypoints, and expectations from the stage in :ref:`stage_table`.
+For example, continuing from the example plugin above::
+
+    entry_points = {
+        'qiskit.transpiler.layout': [
+            'vf2 = qiskit_plugin_pkg.module.plugin:VF2LayoutPlugin',
+        ]
+    },
+
+(note that the entry point ``name = path`` is a single string not a Python
+expression). There isn't a limit to the number of plugins a single package can
+include as long as each plugin has a unique name. So a single package can
+expose multiple plugins if necessary. Refer to :ref:`stage_table` for a list
+of reserved names for each stage.
+
+Plugin API
+==========
+
+.. autosummary::
+   :toctree: ../stubs/
+
+   PassManagerStagePlugin
+   PassManagerStagePluginManager
+   list_stage_plugins
+"""
+
+import abc
+from typing import List
+
+import stevedore
+
+from qiskit.transpiler.passmanager import PassManager
+from qiskit.transpiler.exceptions import TranspilerError
+from qiskit.transpiler.passmanager_config import PassManagerConfig
+
+
+class PassManagerStagePlugin(abc.ABC):
+    """A ``PassManagerStagePlugin`` is a plugin interface object for using custom
+    stages in :func:`transpile()`.
```

Comment:
In the docs, `transpile` is not clickable.

Reply:
This is a typo, this should fix it:
```suggestion
    stages in :func:`~. transpile`.
```